src/test/java/org/jenkinsci/plugins/workflow/cps/DSLTest.java

```diff
@@ -314,6 +315,18 @@ public void namedSoleParamForStep() throws Exception {
         p.setDefinition(new CpsFlowDefinition("try {def c = classLoad(getClass().name); error(/did not expect to be able to load ${c} from ${c.classLoader}/)} catch (ClassNotFoundException x) {echo(/good, got ${x}/)}", false));
         r.assertBuildStatusSuccess(p.scheduleBuild2(0));
     }
+
+    /**
+     *  Test user defined closure execution
+     */
+    @Test public void userDefinedClosureExecution() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        ScriptApproval.get().approveSignature("method groovy.lang.Binding setVariable java.lang.String java.lang.Object");
+        p.setDefinition(new CpsFlowDefinition("binding.setVariable(\"my_closure\", { echo \"my closure!\" })\n my_closure() ", true));
```

Comment:
Nitpick: can just use "false" for the setDefinition argument 2 - that turns off sandbox for the test and you won't need the ScriptApproval.   Seems like a reasonable testcase though.